include/deal.II/distributed/tria.h

```diff
@@ -880,7 +881,37 @@ namespace parallel
         std::map<unsigned int, pack_callback_t> pack_callbacks;
       };
 
-      CellAttachedData cell_attached_data;
+      /**
+       * Vector of tuples containing of p4est quadrant, dealii cells and their relation after refinement.
+       * This makes a one to one correlation possible and allows for parallelization in the future.
+       *
+       * To write its contents, call set_quadrant_cell_pairs(). This vector will be ordered by the occurance of
+       * quadrants in the corresponding p4est sc_array., at which data for transfer will be packed.
+       */
+      std::vector< std::tuple< typename dealii::internal::p4est::types<dim>::quadrant *,
+          typename Triangulation<dim,spacedim>::CellStatus,
+          typename Triangulation<dim,spacedim>::cell_iterator> > local_quadrant_cell_relations;
+
+      /**
+       * Go through all p4est trees and and store them in local_quadrant_cell_pairs.
+       */
+      void store_quadrant_cell_relations();
```

Comment:
store -> compute

how easy would it be to make this function not access the member variable, but instead return an object? Then the function could be `const`.